### PR TITLE
Paralellize the integration tests

### PR DESCRIPTION
**Pull request by hroncok**
This has been a horror.

### CODE REVIEW

**Comment by hroncok** [comment]
Even on Travis Ci this is slightly faster. On my machine, it got from ~9 minutes to ~4.

File: tox.ini [hunk 1]
```diff
@@ -14,13 +14,14 @@ sitepackages = True
 [testenv:integration]
 deps =
     pytest
+    pytest-xdist
     pyyaml
 basepython = python3
-commands = python -m pytest -v {posargs} test/integration
+commands = python -m pytest -v -n3 {posargs} test/integration
```
Comment (hroncok) [inline tox.ini]: Using more is very heavy on disc usage.

File: mock.cfg [hunk 2]
```diff
@@ -1,7 +1,8 @@
-include('/etc/mock/fedora-27-x86_64.cfg')
+include('/etc/mock/fedora-28-x86_64.cfg')
 
 config_opts['chroot_setup_cmd'] = 'install ansible dnf'
 config_opts['use_host_resolv'] = True
 config_opts['rpmbuild_networking'] = True
 config_opts['use_nspawn'] = False
-config_opts['root'] = 'fedora-27-x86_64-taskotron'
+config_opts['root'] = 'taskotron-python-versions-master'
+config_opts['plugin_conf']['root_cache_opts']['dir'] = "%(cache_topdir)s/taskotron-python-versions/root_cache/"
```
Comment (hroncok) [inline mock.cfg]: this makes sure all our mock roots share one root cache

**Comment by hroncok** [comment]
Added an unrelated hotfix. But I was lazy to rebase this on top of it.

**Comment by hroncok** [comment]
@kparal Could you please just tell me whether this works for you and whether it's at least a bit faster?

**Comment by hroncok** [comment]
Thanks @irushchyshyn what happens if you run it for the second time?

**Comment by irushchyshyn** [comment]
> what happens if you run it for the second time?

It was the second time. For the first time it took ~10 mins for integration tests. But I realised I was running it with only 1 CPU :blush:

I increased the number of CPU to 4 (I currently run Fedora in a VM) and it took ~4 min for integration tests (compared to ~8 mins on develop branch ) Twice as fast 👍

**Comment by kparal** [comment]
Before patch:
```
real	10m4.292s
user	7m16.541s
sys	0m30.759s
```
That currently fails in `test_py3_support_failed[bucky]` and `test_artifact_contains_py3_support_and_looks_as_expected`.

After patch:
```
real	5m56.814s
user	12m43.871s
sys	0m54.623s
```
and
```
real	4m3.688s
user	8m51.484s
sys	0m45.015s
```
No crashes.

**Comment by hroncok** [comment]
Thank you both.

**Pull request merged**